client/beefy/src/lib.rs

```diff
@@ -53,13 +53,13 @@ pub(crate) mod beefy_protocol_name {
 	/// Name of the notifications protocol used by BEEFY.
 	///
 	/// Must be registered towards the networking in order for BEEFY to properly function.
-	pub fn standard_name<Hash: std::fmt::Display>(
+	pub fn standard_name<Hash: AsRef<[u8]>>(
 		genesis_hash: &Hash,
 		chain_spec: &Box<dyn ChainSpec>,
 	) -> std::borrow::Cow<'static, str> {
 		let chain_prefix = match chain_spec.fork_id() {
-			Some(fork_id) => format!("/{}/{}", genesis_hash, fork_id),
-			None => format!("/{}", genesis_hash),
+			Some(fork_id) => format!("/{}/{}", hex::encode(genesis_hash), fork_id),
```

Comment:
This branch is not tested because local chain spec doesn't have a `fork_id`.
But since `fork_id` is a simple `&str` there's not much can go wrong when printing it.

If we want to test this as well, I can switch from using local `chain_spec` to defining a mock one with customizable `fork_id` next to all the required traits..

Reply:
Another thing we could do is to replace `chain_spec: &Box<dyn ChainSpec>` in params with `fork_id: Option<&str>`, which also removes the `sc-chain-spec` dependency from GRANDPA and BEEFY clients, _but_ is an API break, implies companion PRs and arguably makes the protocol name easier to misconfigure.